### PR TITLE
replace `delegate_mining` validation to rewards

### PR DESCRIPTION
**Pull request by rwwwx**
replace `delegate_mining` validation to rewards. **let a few todos with questions for reviewers**

### CODE REVIEW

**Comment by rwwwx** [comment]
Also take a look on this https://github.com/adm-metaex/mplx-rewards/pull/38 pr

**Comment by rwwwx** [comment]
It quite raw but I need your review to continue

File: programs/voter-stake-registry/src/cpi_instructions.rs [hunk 1]
```diff
@@ -75,6 +75,8 @@ pub enum RewardsInstruction {
         lockup_period: LockupPeriod,
         /// Specifies the owner of the Mining Account
         owner: Pubkey,
+        /// Wallet addres of delegate
+        delegate_wallet_addr: Pubkey,
```
Comment (StanChe) [inline programs/voter-stake-registry/src/cpi_instructions.rs:79]:
```suggestion
        delegate: Pubkey,
```

File: programs/voter-stake-registry/src/cpi_instructions.rs [hunk 2]
```diff
@@ -348,7 +358,9 @@ pub fn withdraw_mining<'a>(
     amount: u64,
     owner: &Pubkey,
     signers_seeds: &[&[u8]],
+    delegate_wallet_addr: &Pubkey,
 ) -> ProgramResult {
+    // TODO: Maybe it necessary to put `delegate_wallet_addr` into metadata
```
Comment (StanChe) [inline programs/voter-stake-registry/src/cpi_instructions.rs:363]: You will not be reading the `delegate` account on the rewards side. So just passing it as a pubkey is fine.

File: programs/voter-stake-registry/src/lib.rs [hunk 3]
```diff
@@ -234,17 +234,6 @@ impl Stake<'_> {
             MplStakingError::InvalidDelegate
         );
```
Comment (StanChe) [inline programs/voter-stake-registry/src/lib.rs:236]: This method should be renamed to `verify_delegate` as it's not touching the mining any more

File: programs/voter-stake-registry/tests/program_test/rewards.rs [hunk 4]
```diff
@@ -129,6 +129,7 @@ impl RewardsCookie {
         Ok(mining)
     }
 
+    // TODO: Maybe we need to delete this function. It is not used anywhere
```
Comment (kstepanovdev) [inline programs/voter-stake-registry/tests/program_test/rewards.rs:132]: If it's unused, then kill it.

**Review by kstepanovdev — APPROVED** [review]
LGTM

**Review by StanChe — APPROVED** [review]

**Pull request merged**